docs/sphinx/releases/whats_new_1_2_0.rst

```diff
@@ -6,15 +6,271 @@
+
+* Thanks to the work of our Google Summer of Code student, Nikunj Gupta, we now
+  have a new implementation of ``hpx_main.hpp`` on supported platforms (Linux,
+  BSD and MacOS). This is intended to be a less fragile drop-in replacement for
+  the old implementation relying on preprocessor macros. The new implementation
+  does not require changes if you are using the |cmake|_ or pkg-config. The old
+  behaviour can be restored by setting ``HPX_WITH_DYNAMIC_HPX_MAIN=OFF`` during
+  |cmake|_ configuration. The implementation on Windows is unchanged.
+* We have added functionality to allow passing scheduling hints to our
+  schedulers. These will allow us to create executors that for example target a
+  specific NUMA domain or allow for |hpx| threads to be pinned to a particular
+  worker thread.
+* We have significantly improved the performance of our futures implementation
+  by making the shared state atomic.
+* We have replaced Boostbook by Sphinx for our documentation. This means the
+  documentation is easier to navigate with built-in search and table of
+  contents. We have also added a quick start section and restructured the
+  documentation to be easier to follow for new users.
+* We have added a new option to the :option:`--hpx:threads` command line option.
+  It is now possible to use ``cores`` to tell |hpx| to only use one worker
+  thread per core, unlike the existing option ``all`` which uses one worker
+  thread per processing unit (processing unit can be a hyperthread if
+  hyperthreads are available). The default value of :option:`--hpx:threads` has
+  also been changed to ``cores`` as this leads to better performance in most
+  cases.
+* All command line options can now be passed alongside configuration options
+  when initializing |hpx|. This means that some options that were previously
+  only available on the command line can now be set as configuration options.
+* Two new performance counters were added for measuring the time spent doing
+  background work. ``/threads/time/background-work-duration`` returns the time
+  spent doing background on a given thread or locality, while
+  ``/threads/time/background-overhead`` returns the fraction of time spent doing
+  background work with respect to the overall time spent running the scheduler.
+  The new performance counters are disabled by default and can be turned on by
+  setting ``HPX_WITH_BACKGROUND_THREAD_COUNTERS=ON`` during |cmake|_
+  configuration.
+* The idling behaviour of |hpx| has been tweaked to allow for faster idling.
+  This is useful in interactive applications where the |hpx| worker threads may
+  not have work all the time. This behaviour can be tweaked and turned off as
+  before with ``HPX_WITH_THREAD_MANAGER_IDLE_BACKOFF=OFF`` during |cmake|_
+  configuration.
+* It is now possible to register callback functions for |hpx| worker thread
+  events. Callbacks can be registered for starting and stopping worker threads,
+  and for when errors occur.
+
```

Comment:
We should mention the (optional, but experimental) addition of HPXMP as well.

Reply:
Good point! Would a one-sentence description of HPXMP and how to enable it be enough? Would you mind adding that? I think technically you might have to make a PR to my branch since it's not on this repo...

Reply:
Will do.

Reply:
@tianyi93 Would you mind writing a sentence or two explaining what HXPMP is and how to enable its use within HPX?

Reply:
@hkaiser  Sure, I'm happy to do that.

HPXMP is a portabl, scalable and flexible application programming interface using OpenMP specification that supports multi-platform shared memory multiprocessing programming in C/ C++.


HPXMP can be enabled within HPX using the following option -DHPX_WITH_HPXMP=ON

Reply:
Thanks @tianyi93. I'll add this to the PR. Did I see correctly that this essentially just builds libomp (or whatever it's named) with the rest of HPX, but it doesn't add any functionality directly to HPX itself?

Reply:
That's correct.